src/hdf5_interface.F90

```diff
@@ -787,6 +791,76 @@ subroutine hdf5_read_string(group, name, buffer, length)
   end subroutine hdf5_read_string
 
 !===============================================================================
+! HDF5_WRITE_STRING_1DARRAY writes string data
+!===============================================================================
+
+  subroutine hdf5_write_string_1Darray(group, name, buffer, length, rank)
```

Comment:
That all makes sense. I'd say we should leave the new label functionality for a separate PR. What I'm still not seeing though is why this would use write_string_1darray versus just write_string? Where would you actually be writing an array of character variables? That aside, even on the python-api-labels branch, hdf5_write_string_1darray still doesn't actually *write* anything (the code that does so is commented out).